policy-controller/src/admission.rs

```diff
@@ -1,37 +1,74 @@
 use crate::api;
 use anyhow::{anyhow, bail, Result};
 use api::policy::ServerSpec;
+use futures::future;
+use hyper::body::Buf;
+use hyper::http;
+use hyper::{Body, Request, Response};
 use kube::{api::Api, core::DynamicObject, ResourceExt};
-use std::convert::{Infallible, TryInto};
+use std::convert::TryInto;
+use std::{future::Future, pin::Pin};
 use tracing::{debug, info, warn};
-use warp::{filters::BoxedFilter, http, reply, Filter};
 
-/// Builds an API handler for a validating admission webhook that examines `Server` resources.
-pub fn routes(client: kube::Client) -> BoxedFilter<(impl warp::Reply,)> {
-    warp::post()
-        .and(warp::path::end())
-        // 64KB should be more than enough for any `Server` instance we can expect.
-        .and(warp::body::content_length_limit(64 * 1024).and(warp::body::json()))
-        .and(warp::any().map(move || client.clone()))
-        .and_then(|review: Review, client| async move {
-            let req: Request = match review.try_into() {
+type BoxError = Box<dyn std::error::Error + Send + Sync>;
+type ResponseFuture = dyn Future<Output = Result<Response<Body>, BoxError>> + Send + 'static;
+type BoxFuture = Pin<Box<ResponseFuture>>;
+
+#[derive(Clone)]
+pub struct Service {
+    pub client: kube::Client,
+}
+
+impl hyper::service::Service<Request<Body>> for Service {
+    type Response = Response<Body>;
+    type Error = BoxError;
```

Comment:
In the policy-controller, we use the [`anyhow`](https://docs.rs/anyhow/latest/anyhow/index.html) error helpers when we don't care about the error type

```suggestion
    type Error = anyhow::Error;
```